src/transformers/image_processing_base.py

```diff
@@ -95,15 +98,15 @@ def _set_processor_class(self, processor_class: str):
 
     @classmethod
     def from_pretrained(
-        cls,
+        cls: Type[ImageProcessorType],
         pretrained_model_name_or_path: Union[str, os.PathLike],
         cache_dir: Optional[Union[str, os.PathLike]] = None,
         force_download: bool = False,
         local_files_only: bool = False,
         token: Optional[Union[str, bool]] = None,
         revision: str = "main",
         **kwargs,
-    ):
+    ) -> ImageProcessorType:
```

Comment:
This is slightly incorrect because actually, we have `Union[ImageProcessor, Tuple[ImageProcessor, Dict]]`. However, I suppose it's almost never used externally with `return_unused_kwargs=True`, in which case it would return a tuple instead of `ImageProcessor`. Thus, it should improve the experience.